### PR TITLE
Feature/files methods

### PR DESCRIPTION
**Pull request by rrrrs09**
# Description

Added methods for uploading and downloading files. Added fields to `HTTPRequest` and `HTTPResponse` for working with non-json data.

# Checklist

- [x] Linters and tests have passed
- [x] I've written a changelog for PR change
- [x] I'll make a release when PR is approved

### CODE REVIEW

File: botx/clients/clients/processing.py [hunk 1]
```diff
@@ -22,6 +27,12 @@ def extract_result(method: BotXMethod[ResponseT], response: HTTPResponse) -> Res
     Returns:
         Converted shape from BotX API.
     """
+    if method.expected_type == ExpectedType.BINARY:
+        mimetype = response.headers["content-type"].split(";", 1)[0]
+        ext = File.get_ext_by_mimetype(mimetype)
+        file_name = "document{0}".format(ext)
+        return File.from_file(BytesIO(response.raw_data), file_name)  # type: ignore
```
Comment (lxmnk) [inline botx/clients/clients/processing.py:34]: Nit: move it to separate function.

File: botx/clients/types/http.py [hunk 2]
```diff
@@ -24,15 +33,36 @@ class HTTPRequest(BaseModel):
...
+    data: Optional[Dict[str, Any]] = None  # noqa: WPS110
+
+    #: file for httpx in {field_name: (file_name, file_content)}.
+    files: Optional[Dict[str, Tuple[str, BytesIO]]] = None  # noqa: WPS234
+
+    #: expected type of response body.
+    expected_type: ExpectedType = ExpectedType.JSON
+
+    #: extra error codes
+    should_process_as_error: List[int] = []
```
Comment (lxmnk) [inline botx/clients/types/http.py:46]: Do we really need this property? Can we delegate this functionality to `error_handlers`?

File: docs/changelog.md [hunk 3]
```diff
@@ -1,3 +1,20 @@
...
+### Added
+
+* Add method for uploading files to chat.
+* Add method for downloading files from chat.
+
+### Changed
+
+* Add `data` and `files` fields to `HTTPRequest` for sending multipart/form-data in request.
+* Add `expected_type` field to `HTTPRequest` and `expected_type` property to `BaseBotXMethod`
+  to control the construction of the response when the non-json response of the request method.
```
Comment (lxmnk) [inline docs/changelog.md:12]: `to allow interacting with non JSON responses.`

File: botx/bots/mixins/requests/files.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        )
+
+    async def download_file(
+        self: BotXMethodCallProtocol,
+        credentials: SendingCredentials,
+        file_id: UUID,
+        group_chat_id: UUID,
+        *,
+        is_preview: bool = False,
+    ) -> File:
```
Comment (Opti213) [inline botx/bots/mixins/requests/files.py:60]: If this function returns File, you should provide interface to construct undefined fields like filename. Or return binary.

File: botx/clients/clients/sync_client.py [hunk 5]
```diff
@@ -108,19 +108,38 @@ def execute(self, request: HTTPRequest) -> HTTPResponse:
...
+        headers = dict(response.headers)
+
+        should_process_as_error = (
+            response.status_code in request.should_process_as_error
+        )
+        if (  # noqa: WPS337
+            not response.is_error
+            and not should_process_as_error  # noqa: W503
+            and request.expected_type == ExpectedType.BINARY  # noqa: W503
+        ):
```
Comment (Opti213) [inline botx/clients/clients/sync_client.py:129]: Use variable for explain expression. Like IS_FILE_RESPONSE. For `not response.is_error and request.expected_type == ExpectedType.BINARY`

File: botx/clients/clients/sync_client.py [hunk 6]
```diff
@@ -108,19 +108,38 @@ def execute(self, request: HTTPRequest) -> HTTPResponse:
...
             )
         except httpx.HTTPError as httpx_exc:
             raise BotXConnectError(
                 url=request.url,
                 method=request.method,
             ) from httpx_exc
 
+        headers = dict(response.headers)
+
+        should_process_as_error = (
```
Comment (Opti213) [inline botx/clients/clients/sync_client.py:122]: For what? Please, give me case when should we use this?

File: botx/bots/mixins/requests/files.py (hunk 4 above)
Comment (rrrrs09) [reply botx/bots/mixins/requests/files.py:60]: Those fields can be filled directly for the returned file.

File: botx/clients/types/http.py (hunk 2 above)
Comment (rrrrs09) [reply botx/clients/types/http.py:46]: We decided to keep the current version.

File: docs/changelog.md (hunk 3 above)
Comment (rrrrs09) [reply docs/changelog.md]: Done

File: botx/bots/mixins/requests/files.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+        Arguments:
+            credentials: credentials for making request.
+            sending_file: file to upload.
+            group_chat_id: ID of the chat that accepts the file.
+            duration: duration of the voice or the video.
+            caption: file caption.
+
+        Returns:
+            ID of uploaded file.
```
[the remark this answers is not in the document]
Comment (rrrrs09) [reply botx/bots/mixins/requests/files.py]: Yes, done

File: botx/clients/clients/sync_client.py (hunk 6 above)
Comment (rrrrs09) [reply botx/clients/clients/sync_client.py:122]: For handling statuses other than 400 - 599

File: botx/clients/clients/processing.py (hunk 1 above)
Comment (rrrrs09) [reply botx/clients/clients/processing.py]: Done

**Review by lxmnk — APPROVED** [review]

**Review by Nimond — APPROVED** [review]

**Pull request merged**